SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -471,7 +471,9 @@ public void handleCompletedTask(Optional<SingularityTask> task, SingularityTaskI
       stateCache.getActiveTaskIds().remove(taskId);
     }
 
-    taskManager.createLBCleanupTask(taskId);
+    if (task.isPresent() && task.get().getTaskRequest().getRequest().isLoadBalanced()) {
+      taskManager.createLBCleanupTask(taskId);
+    }
```

Comment:
Is this an optimization? Is it 100% safe? What if the task isn't present but it's being load balanced due to a data corruption issue in Singularity?